src/lucky/renderable.cr

```diff
@@ -103,19 +103,66 @@ module Lucky::Renderable
     response.print
   end
 
+  private def handle_response(_response : Nil) forall T
+    # This is a hack to force Crystal to regenerate this method for
+    # each action. That way @type refers to the action with the problem,
+    # and not the first action to be compiled.
+    {% @type %}
+    {%
+      raise <<-ERROR
+
+
+      #{@type} returned Nil
+
+      But it should return a Lucky::Response.
+
+      Try this...
+
+        ▸ Ensure the action returns a response with html/redirect/json/etc.
+        ▸ Ensure all conditionals (like if/else) return a response with html/redirect/json/etc.
+
+      For example...
+
+        get "/admin/users" do
+          # Make sure there is a response in all conditional branches
+          if current_user.admin?
+            html IndexPage, users: UserQuery.new
+          else
+            redirect Home::Index
+          end
+        end
+
+      ERROR
+    %}
+  end
+
   private def handle_response(_response : T) forall T
+    # This is a hack to force Crystal to regenerate this method for
+    # each action. That way @type refers to the action with the problem,
+    # and not the first action to be compiled.
+    {% @type %}
     {%
       raise <<-ERROR
 
 
-      Your action returned #{T}
+      #{@type} returned #{T}
```

Comment:
Isn't this what we had before and it didn't work? 🤔

Reply:
Yeah I was being dumb https://github.com/luckyframework/lucky/pull/1086/files#r403641754

And I couldn't figure out a way to get around it because there was a crystal bug in the compiler. Doing what Ary suggested did work when you didn't return a Lucky::Response, but when you the compiler blew up 🤷‍♂ So I guess this is still not possible. I think I'll keep the other changes and leave the other issue open for one day when we can do what we'd like or figure out a workaround :P